src/apolo_app_types/helm/apps/ingress.py

```diff
@@ -129,14 +130,21 @@ async def get_http_ingress_values(
         **http_ingress_config,  # Merge the generated config directly
     }
 
-    # Handle auth based on its presence in the input object
-    if ingress_http.auth:
+    # Handle auth based on its type
+    if isinstance(ingress_http.auth, ApoloAuth):
         ingress_vals.setdefault("annotations", {})  # Ensure annotations key exists
         is_prod = is_production_cluster(apolo_client)
         middleware_string = _get_middlewares_annotation_value(
```

Comment:
let's maybe drop `_get_middlewares_annotation_value` and use `append_ingress_middleware_annotations` only?